File: src/Atlasd/Battlenet/Protocols/IListener.cs
using System.Net;
using System.Net.Sockets;

namespace Atlasd.Battlenet.Protocols
{
    interface IListener
    {
        public IPEndPoint LocalEndPoint { get; }
        public bool IsListening { get; }
        public Socket Socket { get; }

        public void Close();
        public void SetLocalEndPoint(IPEndPoint endp);
        public void Start();
        public void Stop();
    }
}

File: src/AtlasV/Battlenet/Protocols/HTTP/IListener.vb
Imports System.Net
Imports System.Net.Sockets

Namespace AtlasV.Battlenet.Protocols
    Interface IListener
        Property LocalEndPoint() As IPEndPoint
        ReadOnly Property IsListening As Boolean
        Property Socket As Socket
        Sub Close()
        Sub SetLocalEndPoint(ByVal varEndPoint As IPEndPoint)
        Sub Start()
        Sub [Stop]()
    End Interface
End Namespace
